verifier/verifier.go

```diff
@@ -0,0 +1,133 @@
+/*
+ * ZCrypto Copyright 2017 Regents of the University of Michigan
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package verifier
+
+import (
+	"time"
+
+	"github.com/zmap/zcrypto/x509"
+)
+
+// VerificationResult contains the result of a verification of a certificate
+// against a store and its associated policies. It discerns between NameError,
+// meaning the name on the certificate does not match, and a ValidationError,
+// meaning some sort of issue with the cryptography, or other issues with the
+// chain.
+type VerificationResult struct {
+
+	// The name being checked, e.g. "www.censys.io".
+	Name string
+
+	// Whitelisted is true if the certificate is whitelisted by a given verifier
+	// as trusted, e.g. when Apple whitelisted subset of StartCom certs by SPKI
+	// hash in the wake of the WoSign incidents surrounding misissuance.
+	Whitelisted bool
```

Comment:
Maybe a slice of errors, rather than 3 independent booleans and two errors? I haven't seen that pattern before, but I also dislike looking in N places for a validation result